forecasting/src/autogluon/forecasting/trainer/abstract_trainer.py

```diff
@@ -0,0 +1,668 @@
+        the validation score."""
+
+        if model is None:
+            logger.log(
+                30,
+                "Model not specified, will default to the model with the best validation score",
+            )
+            if self.model_best is None:
+                best_model_name: str = self.get_model_best()
+                self.model_best = best_model_name
+            return self.load_model(self.model_best)
+        else:
+            if isinstance(model, str):
+                return self.load_model(model)
+            return model
+
+    def predict(
+        self,
+        data: TimeSeriesDataFrame,
+        model: Optional[AbstractForecastingModel] = None,
+        **kwargs,
+    ) -> TimeSeriesDataFrame:
+        model = self._get_model_for_prediction(model)
+        return self._predict_model(data, model, **kwargs)
+
+    def score(
+        self,
+        data: TimeSeriesDataFrame,
+        model: Optional[Union[str, AbstractModel]] = None,
+        quantiles: List[float] = None,
+    ) -> float:
+        evaluator = Evaluator(quantiles=quantiles or self.quantiles)
+        model = self._get_model_for_prediction(model)
+
+        # FIXME: this method should be able to score on all data sets regardless of
+        # FIXME: whether the implementation is in GluonTS
+        if not isinstance(model, AbstractGluonTSModel):
+            raise ValueError("Model must be a GluonTS model to predict for scoring")
+        forecasts, tss = model._predict_for_scoring(data)  # noqa
+        num_series = len(tss)
+
+        with evaluator_warning_filter():
+            agg_metrics, item_metrics = evaluator(
+                iter(tss), iter(forecasts), num_series=num_series
+            )
+        return agg_metrics[self.eval_metric]
+
+    def _predict_model(
+        self,
+        data: TimeSeriesDataFrame,
+        model: Union[str, AbstractForecastingModel],
+        **kwargs,
+    ) -> TimeSeriesDataFrame:
+        if isinstance(model, str):
+            model = self.load_model(model)
+        return model.predict(data, **kwargs)
+
+    # TODO: experimental
+    def refit_single_full(self, train_data=None, val_data=None, models=None):
+        warn("Refitting logic is experimental for autogluon.forecasting.")
+        models_trained_full = []
+        model_full_dict = {}
+
+        train_data = train_data or self.load_train_data()  # noqa
+        val_data = val_data or self.load_val_data()
+
+        # FIXME: implement the append operation below in datasets
+        # refit_full_data = train_data + val_data
+        # for now we assume validation data includes train data (as in GluonTS)
+        refit_full_data = val_data
+
+        if models is None:
+            self.get_model_names_all()
+
+        for model in models:
+            model: AbstractForecastingModel = self.load_model(model)
+            model_name = model.name
+            model_full = model.convert_to_refit_full_template()  # FIXME: not available
```

Comment:
What is reason for it not being available? Is there complexity in implementation?

Reply:
Not particularly, but we do not currently support `params_trained` and incremental training. I know this is important functionality but I'd like to address this in a future PR.

Reply:
Ok, we can add in follow-up PR.